libraries/AP_DroneCAN/AP_DroneCAN_DNA_Server.cpp

```diff
@@ -61,53 +65,42 @@ void AP_DroneCAN_DNA_Server::getHash(NodeData &node_data, const uint8_t unique_i
     }
 }
 
-//Read Node Data from Storage Region
-bool AP_DroneCAN_DNA_Server::readNodeData(NodeData &data, uint8_t node_id)
+// compute safe checksum of NodeData structure
+uint8_t AP_DroneCAN_DNA_Server::getNodeDataChecksum(NodeData &node_data) const
 {
-    if (node_id > MAX_NODE_ID) {
-        return false;
-    }
-    WITH_SEMAPHORE(storage_sem);
-    if (!storage.read_block(&data, (node_id * sizeof(struct NodeData)) + NODEDATA_MAGIC_LEN, sizeof(struct NodeData))) {
-        //This will fall through to Prearm Check
-        server_state = STORAGE_FAILURE;
-        return false;
-    }
-    return true;
+    uint8_t crc = crc_crc8(node_data.hwid_hash, sizeof(node_data.hwid_hash));
+    // 0 signifies empty data record so we have to return 1 instead
+    return crc != 0 ? crc : 1;
```

Comment:
i think we should consider an entry to be invalid only if all of hwid_hash is zero

Reply:
Couldn't find a zero checker function, just used memcmp.